packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts

```diff
@@ -767,6 +767,12 @@ export interface IUserPool extends IResource {
    */
   readonly userPoolArn: string;
 
+  /**
+   * User pool provider name
+   * @attribute
+   */
+  readonly userPoolProviderName: string;
```

Comment:
Is there anything that would prevent us from making this new attribute optional for the interface and base class instead? I'm thinking of the case of users who may be either implementing this interface or extending the abstract class, so adding a new mandatory field would likely break their implementations.

Reply:
Update: I have since looked a little more into these sorts of changes. We don't currently support extending interfaces or abstract classes, so adding a mandatory attribute here isn't a problem!